Foundation/src/PatternFormatter.cpp

```diff
@@ -81,9 +83,13 @@ void PatternFormatter::format(const Message& msg, std::string& text)
 		case 'T': text.append(msg.getThread()); break;
 		case 'I': NumberFormatter::append(text, msg.getTid()); break;
 		case 'J': NumberFormatter::append(text, msg.getOsTid()); break;
-		case 'N': text.append(Environment::nodeName()); break;
+		case 'N':
+			if (_cachedNodeName.empty())
+				_cachedNodeName = Environment::nodeName();
```

Comment:
Logged node name will be inaccurate if node name changes while the application is running. Shall node name be cached for limited time?